source/docs/software/troubleshooting/robots-shouldnt-quit.rst

```diff
@@ -0,0 +1,415 @@
+                  at frc.robot.Robot.fooInit(Foo.java:34)
+                  at frc.robot.Robot.robotInit(Robot.java:24)
+                  at edu.wpi.first.wpilibj.TimedRobot.startCompetition(TimedRobot.java:94)
+                  at edu.wpi.first.wpilibj.RobotBase.runRobot(RobotBase.java:335)
+                  at edu.wpi.first.wpilibj.RobotBase.lambda$startRobot$0(RobotBase.java:387)
+                  at java.base/java.lang.Thread.run(Thread.java:834)
+
+      In this case: ``robotInit`` called ``fooInit``, which in turn called ``barInit``, which in turn called ``buggyMethod``. Then, during the execution of ``buggyMethod``, the ``NullPointerException`` occurred.
+
+   .. group-tab:: C++
+
+      Coming Soon!
+
+
+Perform Code Analysis
+^^^^^^^^^^^^^^^^^^^^^
+
+Once you've found the stack trace, and found the line of code which is triggering the unhandled exception, you can start the process of determining root cause.
+
+Often, just looking at (or near) the problematic line of code will be fruitful. You may notice things you forgot, or lines which don't match an example you're referencing. 
+
+.. note:: Developers who have lots of experience working with code will often have more luck looking at code than newer folks. That's ok, don't be discouraged! The experience will come with time.
+
+A key strategy for analyzing code is to ask the following questions:
+
+ * When was the last time the code "worked" (IE, didn't have this particular error)?
+ * What has changed in the code between the last working version, and now?
+
+Frequent testing and careful code changes help make this particular strategy more effective. 
+
+Run the Single Step Debugger
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Sometimes, just looking at code isn't enough to spot the issue. The :ref:`single-step debugger <docs/software/vscode-overview/debugging-robot-program:Debugging a Robot Program>` is a great option in this case - it allows you to inspect the series of events *leading up to* the unhandled exception.
+
+Search for More Information
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+`Google <https://google.com>`_ is a phenomenal resource for understanding the root cause of errors. Searches involving the programming language and the name of the exception will often yield good results on more explanations for what the error means, and how it often comes about.
+
+Seeking Outside Help
+^^^^^^^^^^^^^^^^^^^^
+
+If all else fails, you can seek out advice and help from others (both in-person and online). When working with folks who aren't familiar with your codebase, it's very important to provide the following information:
+
+ * Access to your source code, (EX: :ref:`on github.com <docs/software/basic-programming/git-getting-started:Git Version Control Introduction>`)
+ * The **full text** of the error, including the full stack trace.
+
+Common Examples & Patterns
+--------------------------
+
+There are a number of common issues which result in runtime exceptions. 
+
+``Null``/``NULL``

```

Comment:
If possible, this might be better as:
```suggestion
``null``/``NULL``
```

Reply:
Lemme go look, this might get resolved as part of the C++ expansion. You got it though, the intent was to cover Java's `null` and C++'s `NULL`. I'm thinking an `inline code block containing NULL` will indicate the programming-language specific identifier, while plain old "null" will indicate the general concept.

Reply:
Ah, okay. In any case, if you want to talk about the concept of null values, that wouldn't be in a code tag, e.g.
> Null/`null`/`NULL`
although I personally would prefer just `null`/`NULL`.